compiler/src/dmd/dsymbolsem.d

```diff
@@ -814,6 +814,19 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             }
         }
 
+        /* If the alignment of a stack local is greater than the stack alignment,
+         * note it in the enclosing function's alignSectionVars
+         */
+        if (!dsym.alignment.isDefault() && sc.func &&
+            dsym.alignment.get() > target.stackAlign() &&
+            sc.func && !dsym.isDataseg() && !dsym.isParameter() && !dsym.isField())
+        {
+            auto fd = sc.func;
+            if (!fd.alignSectionVars)
+                fd.alignSectionVars = new VarDeclarations();
+            fd.alignSectionVars.push(dsym);
+        }
```

Comment:
`version (MARS) { }` block, else there'll be undefined symbol errors building gdc and ldc.

Reply:
> we should test these additions in the cxxfrontend test

Since the MARS frontend has no use for cxxfrontend, there's no point to it.